packages/governance/src/types.js

```diff
@@ -87,74 +151,101 @@
+ *   hold tightly. submitVote() is the core capability that allows the holder to
+ *   specify the identity and choice of a voter. The voteCounter is making that
+ *   available to the Registrar, which should wrap and attenuate it so each
+ *   voter gets only the ability to cast their own vote at a weight specified by
+ *   the registrar.
+ *   attenuated wrapper that doesn't make that available!
+ * @property {SubmitVote} submitVote
  */
 
 /**
- * @typedef {Object} BallotCounterPublicFacet
+ * @typedef {Object} VoteCounterPublicFacet
  * @property {() => boolean} isOpen
- * @property {() => Ballot} getBallotTemplate
- * @property {() => Promise<string>} getOutcome
+ * @property {() => Question} getQuestion
+ * @property {() => Promise<Position>} getOutcome
+ * @property {() => QuestionDetails} getDetails
  * @property {() => Promise<VoteStatistics>} getStats
  */
 
 /**
- * @typedef {Object} BallotCounterCloseFacet
+ * @typedef {Object} VoteCounterCloseFacet
  *   TEST ONLY: Should not be allowed to escape from contracts
  * @property {() => void} closeVoting
  */
 
 /**
- * @typedef {Object} CompleteEqualWeightBallot
- * @property {string} question
- * @property {string[]} chosen - a list of equal-weight preferred positions
+ * @typedef {Object} VoteCounterFacets
+ * @property {VoteCounterPublicFacet} publicFacet
+ * @property {VoteCounterCreatorFacet} creatorFacet
+ * @property {VoteCounterCloseFacet} closeFacet
  */
 
 /**
- * @typedef {Object} CompleteWeightedBallot
- * @property {string} question
- * @property {Record<string,bigint>[]} weighted - list of positions with weights.
- *   BallotCounter may limit weights to a range or require uniqueness.
+ * @callback BuildVoteCounter
+ * @param {QuestionSpec} questionSpec
+ * @param {bigint} threshold - questionSpec includes quorumRule; the registrar
+ *    converts that to a number that the counter can enforce.
+ * @param {Instance} instance
+ * @returns {VoteCounterFacets}
  */
 
 /**
- * @typedef {Object} CompleteOrderedBallot
- * @property {string} question
- * @property {string[]} ordered - ordered list of position from most preferred to
- *   least preferred
+ * @callback LooksLikeQuestionSpec
+ * @param {QuestionSpec} allegedQuestionSpec
+ * @returns {QuestionSpec}
  */
 
 /**
- * @typedef { CompleteEqualWeightBallot | CompleteOrderedBallot | CompleteWeightedBallot } CompletedBallot
+ * @callback LooksLikeIssueForType
+ * @param {ElectionType} electionType
+ * @param {Issue} issue
  */
 
 /**
  * @callback SubmitVote
- * @param {Handle<'Voter'>} seat
- * @param {ERef<CompletedBallot>} filledBallot
+ * @param {Handle<'Voter'>} voterHandle
+ * @param {[Position]} chosenPositions
  * @param {bigint=} weight
  */
 
 /**
- * @typedef {Object} VoterFacet
- * @property {SubmitVote} submitVote
+ * @typedef {Object} RegistrarPublic
+ * @property {() => Subscription<QuestionDetails>} getQuestionSubscription
+ * @property {() => ERef<Handle<'Question'>[]>} getOpenQuestions,
```

Comment:
`ERef<T>` makes sense on arguments where we allow either T or `Promise<T>` but for return types, we know we're returning a `Promise<T>`, no?

```suggestion
 * @property {() => Promise<Handle<'Question'>[]>} getOpenQuestions,
```

Reply:
done